src/lib/comms/include/sol-connman.h

```diff
@@ -0,0 +1,285 @@
+ * Since the connman function is asynchronous, the call return is not connection error,
+ * but just some dispatching/immediate error. The actual state change will be notified
+ * via sol_connman_add_service_monitor() callbacks.
+ * After the actual state change is notified via sol_connman_add_service_monitor()
+ * callbacks, sol_connman_service_get_call_result can be used to get the actual state.
+ *
+ * @return TRUE on success, FALSE on failure
+ */
+bool sol_connman_service_get_call_result(
+    const struct sol_connman_service *service);
+
+/**
+ * @brief Gets the global connection state of system
+ *
+ * This function gets the global connection state for system.
+ *
+ * @return The state of the global connection state on success,
+ * SOL_CONNMAN_STATE_UNKNOWN on failure
+ */
+enum sol_connman_state sol_connman_get_state(void);
+
+/**
+ * @brief Sets the global connection state to offline
+ *
+ * This function sets the global connection state to offline,
+ * enter offline Mode too.
+ *
+ * @param enabled The value is set for offline on/off
+ *
+ * @return 0 on success, -errno on failure.
+ */
+int sol_connman_set_offline(bool enabled);
+
+/**
+ * @brief Gets the global connection state of offline
+ *
+ * This function gets the global connection state of offline.
+ *
+ * @return True if the offline is enabled, false otherwise.
+ */
+bool sol_connman_get_offline(void);
+
+/**
+ * @brief Connect the service
+ *
+ * This function connects the service, after the service is to give by service monitor.
+ * Since the connman function is asynchronous, the return is not connection error,
+ * but just some dispatching/immediate error. The actual state change will be notified
+ * via sol_connman_add_service_monitor() callbacks. This callback must be added BEFORE
+ * the sol_connman_service_connect() is called to ensure no messages are lost.
+ *
+ * @param service The service structure which the connection is desired
+ *
+ * @return 0 on success, -errno on failure.
+ */
+int sol_connman_service_connect(struct sol_connman_service *service);
+
+/**
+ * @brief Disconnect the service
+ *
+ * This function disconnects the service.
+ * Since the connman function is asynchronous, the return is not disconnection error,
+ * but just some dispatching/immediate error. The actual state change will be notified
+ * via sol_connman_add_service_monitor() callbacks. This callback must be added BEFORE
+ * the sol_connman_service_disconnect() is called to ensure no messages are lost.
+ *
+ * @param service The service structure which the disconnection is desired
+ *
+ * @return 0 on success, -errno on failure.
+ */
+int sol_connman_service_disconnect(struct sol_connman_service *service);
+
+/**
+ * @brief Adds a monitor for the updated connman services
+ *
+ * To monitor the property of services, it gives information
+ * about the services. The callback is used to provide the information.
+ * sol_connman_add_service_monitor will call sol_connman_lazy_init
```

Comment:
No need to document implementation details. The API documentation should tell the user all the information he/she needs to use these functions properly, not how they work under the hood.